### PR TITLE
feat(components): add 3 react components (with tests)

### PR DESCRIPTION
**Pull request by olliva**
Components: messageBox, messageList, messageItem.
First test for messageBox.

Related: #6

### CODE REVIEW

**Comment by Faraser** [comment]
:+1:

File: app/app.jsx [hunk 1]
```diff
@@ -1,12 +1,26 @@
 import React from 'react'
 import Index from 'components/index'
+import MessageList from './components/messageList/index.jsx';
+
+import MessageBox from 'components/messageBox/index.jsx'
```
Comment (dimaip) [inline app/app.jsx]: Why doesn't this start with `./` like MessageList?

File: package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "description": "chat",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "test": "mocha --compilers js:babel-core/register --require ./app/tests/test_helper.js './app/tests/**/*.@(js|jsx)'",
```
Comment (dimaip) [inline package.json]: It's better to rename it to `test-client`, as there also will be `test-server`.
Also it's better to not depend on global presence of mocha and use local version of it: `./node_modules/mocha/bin/mocha`

Comment (dimaip) [inline package.json]: Also could be nice to add smth like: `"test-client:watch": "npm run test-client -- --watch",`

File: server/server.js [hunk 3]
```diff
@@ -2,6 +2,7 @@
 import React from 'react';
 import express from 'express';
 import path from 'path';
+
```
Comment (dimaip) [inline server/server.js]: Strange addition to server side code :)

File: app/tests/components/MessageBox.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+const {renderIntoDocument, scryRenderedDOMComponentsWithTag, Simulate} = React.addons.TestUtils;
+
+console.log('ms', MessageBox)
+
+describe('MessageBox', () => {
+
+    // ...
+
+    it('invokes callback when a button is clicked', () => {
```
Comment (dimaip) [inline app/tests/components/MessageBox.js]: Is this really what this test tests?
Looking at the test it looks like the more appropriate name would be 'clears message box after sending message'.
Also we could add more tests later on, to test other aspects of this component.

File: app/tests/test_helper.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import jsdom from 'jsdom';
```
Comment (dimaip) [inline app/tests/test_helper.js]: At server-side we call the dir with tests `test`: https://github.com/shri-2015-org/shrimp/tree/master/server/test
Even though I like `tests` better, we should stick to `test` for consistency, or rename in both places.

**Comment by dimaip** [comment]
Great work! Left a few tiny comments that need to be fixed. Also this branch must be rebased before merging.
:-1:

File: app/tests/components/MessageBox.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import React from 'react/addons';
+import MessageBox from 'components/message-box';
+import {expect} from 'chai';
+import {describe, it} from 'mocha';
```
Comment (SeqviriouM) [inline app/tests/components/MessageBox.js]: I think it doesn't necessarry to import {describe, it} from mocha, because you run them by mocha

Comment (Nitive) [inline app/tests/components/MessageBox.js]: No, you must import this function if you want to use it

**Pull request merged**